### PR TITLE
go.mod: update kubernetes and its deps to v1.17.2

### PR DESCRIPTION
**Pull request by marquiz**
Also syncs e2e tests with the Kubernetes e2e test framework.

### CODE REVIEW

**Comment by zvonkok** [comment]
@marquiz The backward compatibility for the API resources would be >=1.15, right? 
Do we have somewhere stated which version of NFD works with which version of Kubernetes? 
Going forward we might also update `apiVersion` of all resources used to deploy NFD.

File: go.mod [hunk 1]
```diff
@@ -3,96 +3,55 @@ module sigs.k8s.io/node-feature-discovery
 go 1.12
```
Comment (ArangoGutierrez) [inline go.mod]:
```suggestion
go 1.13
```

**Comment by ArangoGutierrez** [comment]
closes #281

**Comment by zvonkok** [comment]
@ArangoGutierrez thanks for testing, see comment https://github.com/kubernetes-sigs/node-feature-discovery/issues/281#issuecomment-582548851 

/lgtm

**Pull request merged**